### PR TITLE
feat: set token amount to balance when clicking the balance

### PR DESCRIPTION
**Pull request by Diogomartf**
Akash gave a good suggestion about setting the balance to max when clicking the balance.

This PR implements that.

https://github.com/SwaprHQ/stackly-ui/assets/5664434/cdbd05fc-a82f-4ca1-a768-e9c9911b3903

### CODE REVIEW

File: packages/app/components/stackbox/Stackbox.tsx [hunk 1]
```diff
@@ -222,27 +232,24 @@ export const Stackbox = () => {
...
                 <TokenIcon token={fromToken} size="2xs" />
-                <BodyText className="text-em-high">
-                  <span className="text-em-low">Balance:</span>{" "}
-                  {formattedBalance(balance)}
-                </BodyText>
+                <div className="flex items-center space-x-1">
+                  <BodyText className="flex items-center text-em-low">
+                    Balance:
+                  </BodyText>
+                  <div onClick={() => setTokenAmountBasedOnBalance(1)}>
```
Comment (berteotti) [inline packages/app/components/stackbox/Stackbox.tsx:247]: Maybe we should have a name for this 1. `Balance.MAX` for example

Comment (Diogomartf) [reply packages/app/components/stackbox/Stackbox.tsx:247]: let's name them all.

Comment (berteotti) [reply packages/app/components/stackbox/Stackbox.tsx:247]: Yes, if we name 1 we name them all :)

File: packages/app/components/stackbox/Stackbox.tsx [hunk 2]
```diff
@@ -52,10 +52,16 @@ const startDateTimeTimestamp = new Date().setMinutes(
   new Date().getMinutes() + 30
 );
 
+const BalanceDivider = {
+  max: 1,
+  half: 2,
+  quarter: 4,
```
Comment (berteotti) [inline packages/app/components/stackbox/Stackbox.tsx:58]: Can we make it uppercase to show that's a constant? Was thinking about using an enum but this works fine too.

Comment (Diogomartf) [reply packages/app/components/stackbox/Stackbox.tsx:58]: I can.

Comment (Diogomartf) [reply packages/app/components/stackbox/Stackbox.tsx]: fixed!

**Review by berteotti — APPROVED** [review]
Good job Diogo!

File: packages/app/components/stackbox/Stackbox.tsx [hunk 3]
```diff
@@ -214,35 +230,36 @@ export const Stackbox = () => {
...
-                </BodyText>
+                <div className="flex items-center space-x-1">
+                  <BodyText className="flex items-center text-em-low">
+                    Balance:
+                  </BodyText>
+                  <div
+                    onClick={() =>
+                      setTokenAmountBasedOnBalance(BalanceDivider.MAX)
+                    }
+                  >
```
Comment (berteotti) [inline packages/app/components/stackbox/Stackbox.tsx:257]: We shouldn't use divs with onClick, this should be a button.

**Review by berteotti — CHANGES_REQUESTED** [review]
Forgot to check the button

File: packages/app/components/stackbox/Stackbox.tsx (hunk 3 above)
Comment (berteotti) [reply packages/app/components/stackbox/Stackbox.tsx:257]: Also, we can declare the button inside the `BodyText` and just apply a `className="cursor-pointer hover:underline text-em-high"` to it.

**Review by berteotti — APPROVED** [review]

**Pull request merged**